doctor_appointment_frontend/src/components/add.js

```diff
@@ -0,0 +1,191 @@
+
+  const handleNameChange = (event) => {
+    setName(event.target.value);
+    setData({ ...data, name: event.target.value });
+  };
+
+  const handleSpecialityChange = (event) => {
+    setSpeciality(event.target.value);
+    setData({ ...data, speciality: event.target.value });
+  };
+
+  const handleDescriptionChange = (event) => {
+    setDescription(event.target.value);
+    setData({ ...data, description: event.target.value });
+  };
+
+  const handleExperienceChange = (event) => {
+    setExperience(event.target.value);
+    setData({ ...data, experience: event.target.value });
+  };
+
+  const handleFeesChange = (event) => {
+    setFees(event.target.value);
+    setData({ ...data, consultation_fees: event.target.value });
+  };
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    if (selectedFile instanceof File) {
+      const formData = new FormData();
+      formData.append('doctor[name]', name);
+      formData.append('doctor[speciality]', speciality);
+      formData.append('doctor[description]', description);
+      formData.append('doctor[consultation_fees]', fees);
+      formData.append('doctor[years_of_experience]', experience);
+      formData.append('doctor[image]', selectedFile, selectedFile.name);
+
+      try {
+        await axios.post(
+          'http://localhost:3000/api/v1/doctors',
+          formData,
+          {
+            headers: {
+              'Content-Type': 'multipart/form-data',
+              'Access-Control-Allow-Origin': 'http://localhost:3000',
+            },
+          },
+        );
+      } catch (error) {
+        /* empty */
+      }
+    }
+  };
+
+  return (
+    <div
+      className="h-screen reservation-container"
+      style={{ backgroundImage: `url(${doc5})`, backgroundSize: 'cover' }}
+    >
+      <div className="relative flex h-screen w-full flex-col items-center justify-center text-white reservation">
+        <h1 className="mb-6 text-3xl font-bold tracking-2xl space-x-1">
+          ADD A DOCTOR
+          {' '}
+        </h1>
+        <hr className="w-1/2 text-white" />
+        <form
+          onSubmit={handleSubmit}
+          className="mx-auto mt-3 w-full max-w-sm grid-cols-2 sm:grid"
+        >
+          <div className="relative mb-4 flex items-center justify-center">
+            <input
+              id="name"
+              name="name"
+              value={name}
+              required
+              placeholder="Name"
+              onChange={handleNameChange}
+              className="whiteInp w-3/4 placeholder:text-white appearance-none rounded border bg-transparent p-4 py-2 leading-tight text-white focus:shadow-outline focus:outline-none"
+            />
+          </div>
+
+          <div className="relative mb-4 flex text-white items-center justify-center">
+            <input
+              id="speciality"
+              name="speciality"
+              value={speciality}
+              placeholder="Speciality"
+              required
```

Comment:
These inputs are very similar, maybe they can be abstracted with subcomponent, array, and map [OPTIONAL]